compendium/DeclarativeServices/src/manager/ConfigurationManager.cpp

```diff
@@ -36,12 +36,15 @@ namespace cppmicroservices
             , metadata(metadata)
             , bundleContext(bc)
             , mergedProperties(metadata->properties)
-            , changeCount(0)
         {
             if (!this->metadata || !this->bundleContext || !this->logger)
             {
                 throw std::invalid_argument("ConfigurationManagerImpl - Invalid arguments passed to constructor");
             }
+            for (auto const& pid : metadata->configurationPids)
+            {
+                changeCount[pid] = 0;
+            }
```

Comment:
is this necessary? doesn't the `unordered_map` guarantee zero initialized values if the key doesn't exist?

Reply:
yes you're correct, I verified this statement and it does return 0